firefox-ios/WidgetKit/DownloadManager/DownloadLiveActivity.swift

```diff
@@ -83,3 +96,87 @@ struct DownloadLiveActivity: Widget {
         }
     }
 }
+
+@available(iOS 16.2, *)
+@DynamicIslandExpandedContentBuilder
+private func expandedContent
+(liveDownload: ActivityViewContext<DownloadLiveActivityAttributes>, settings: expandedContentSettings) ->
+DynamicIslandExpandedContent<some View> {
+        DynamicIslandExpandedRegion(.leading) {
+            ZStack {
+                RoundedRectangle(cornerRadius: settings.iconEdgeRounding)
+                    .fill(settings.widgetColours)
+                    .frame(width: settings.iconFrameSize,
+                           height: settings.iconFrameSize)
+                Image(settings.firefoxIcon)
+                    .resizable()
+                    .scaledToFit()
+                    .frame(width: settings.firefoxIconSize,
+                           height: settings.firefoxIconSize)
+            }.padding(EdgeInsets(top: settings.iconTopPadding,
+                                 leading: settings.iconLeftPadding,
+                                 bottom: settings.iconBottomPadding,
+                                 trailing: settings.iconRightPadding))
+        }
+        DynamicIslandExpandedRegion(.trailing) {
+            ZStack {
+                Circle()
+                    .stroke(settings.widgetColours.opacity(settings.inProgessOpacity),
+                            lineWidth: settings.progressWidth)
+                    .frame(width: settings.iconFrameSize,
+                           height: settings.iconFrameSize)
+                Circle()
+                    .trim(from: 0.0,
+                          to: min(liveDownload.state.totalProgress, 1.0))
+                    .stroke(style: StrokeStyle(lineWidth: settings.progressWidth))
+                    .rotationEffect(.degrees(270.0))
+                    .animation(.linear, value: 0.5)
+                if liveDownload.state.totalProgress == 1.0 {
+                    Image(settings.checkmarkIcon)
+                    .renderingMode(.template)
+                    .frame(width: settings.stateIconSize,
+                           height: settings.stateIconSize)
+                    .foregroundStyle(settings.widgetColours)
+                } else {
+                    Image(settings.mediaStopIcon)
+                        .resizable()
+                        .scaledToFit()
+                        .frame(width: settings.stateIconSize,
+                               height: settings.stateIconSize)
+                }
```

Comment:
We can simplify this a bit since both icons will have the same size 
```suggestion
                Image(
                    liveDownload.state.totalProgress == 1.0
                    ? UX.checkmarkIcon
                    : UX.mediaStopIcon
                )
                .resizable()
                .scaledToFit()
                .renderingMode(.template)
                .foregroundStyle(UX.widgetColours)
                .frame(width: UX.stateIconSize, height: UX.stateIconSize)
```